File: src/org/jetbrains/agda/core/CRefExpression.java
package org.jetbrains.agda.core;

import com.intellij.psi.PsiElement;

public class CRefExpression extends CExpression {
    private PsiElement myDeclaration;
    private String myText;

    public CRefExpression(PsiElement declaration, String text) {
        myDeclaration = declaration;
        myText = text;
    }

    @Override
    public String toString() {
        return myText;
    }
}

File: src/org/jetbrains/agda/core/expression/CRefExpression.kt
package org.jetbrains.agda.core

import com.intellij.psi.PsiElement

/**
 * @author Evgeny.Kurbatsky
 */
public class CRefExpression(val declaration : Any, val  name: String) : CExpression() {

    fun equals(other : Any) : Boolean {
        if (other is CRefExpression) {
            return other.declaration.equals(declaration);
        }
        return false;
    }

    fun toString() : String {
        return name
    }



}
